music21/spanner.py

```diff
@@ -752,10 +803,17 @@ def getLast(self):
 
 
 # ------------------------------------------------------------------------------
-class _SpannerRef(t.TypedDict):
+class PendingAssignmentRef(t.TypedDict):
+    '''
+    An object containing information about a pending first spanned element
+    assignment. See setPendingFirstSpannedElementAssignment for documentation
+    and tests.
+    '''
     # noinspection PyTypedDict
     spanner: 'Spanner'
     className: str
+    offsetInScore: OffsetQL|None
+    clientInfo: t.Any|None
```

Comment:
I still have no idea what `clientInfo` is for or why it should be here.  `t.Any` seems like a bad typing.

Reply:
In the next PR, xmlToM21.py uses it to stash off the staffKey for each pending assignment (staffKey is used when splitting a Part into multiple PartStaffs).  When it finds pending assignments that were not ever resolved (by calling popPendingSpannedElementAssignments), it creates a SpannerAnchor for each one, using that stashed off staffKey to put the anchor in the correct split-out staff (of a piano score, for example).  I could name/type it as staffKey: int (and in fact I did for a while), but that seems very specific to one client.  "clientInfo: t.Any|None" makes it so that any client can put anything they need to remember about this particular pending assignment there, and then read it back out after calling popPendingSpannedElementAssignments.

Reply:
This is a classic (a.k.a. real old, like me) software engineering pattern.  We used to call clientInfo "refcon".

Reply:
Switched this to staffKey: int|None as requested.